### ISSUE TITLE
Error "csc2.exe" exited with code -1073741568.

### ISSUE THREAD

**Issue by vasily-kirichenko** [dotnet/roslyn#969]
1. Try to build Hopac.Core project from Hopac solution from here https://github.com/Hopac/Hopac (all the other project in that solution are F# ones, so you need Visual F# Tools installed to open the solution). 
2. C# compiler fails with `Error "csc2.exe" exited with code -1073741568. Hopac.Core` 

It's successfully compiled with VS 2015 CTP 5, but is not with CTP 6.

**Comment by tmeschter** [comment]
@vasily-kirichenko Try adding this property:

`<CscToolExe>csc.exe</CscToolExe>`

to the project file and build again. That may turn up more useful error information.

**Comment by pdelvo** [comment]
Those high error codes are hinting at a managed exception. I tried to compile it, it crashed and I looked at the event viewer. Its a NullReferenceException:
```
Meldung: System.NullReferenceException: Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt.
   bei Microsoft.CodeAnalysis.CSharp.BoundFieldAccess.NeedsByValueFieldAccess(BoundExpression receiver, FieldSymbol fieldSymbol)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindFieldAccess(CSharpSyntaxNode node, BoundExpression receiver, FieldSymbol fieldSymbol, DiagnosticBag diagnostics, LookupResultKind resultKind, Boolean hasErrors)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindNonMethod(SimpleNameSyntax node, Symbol symbol, DiagnosticBag diagnostics, LookupResultKind resultKind, Boolean isError)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindIdentifier(SimpleNameSyntax node, Boolean invoked, DiagnosticBag diagnostics)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindLeftOfPotentialColorColorMemberAccess(ExpressionSyntax left, DiagnosticBag diagnostics)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindMemberAccess(MemberAccessExpressionSyntax node, Boolean invoked, Boolean indexed, DiagnosticBag diagnostics)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindInvocationExpression(InvocationExpressionSyntax node, DiagnosticBag diagnostics)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindExpressionInternal(ExpressionSyntax node, DiagnosticBag diagnostics, Boolean invoked, Boolean indexed)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindExpression(ExpressionSyntax node, DiagnosticBag diagnostics, Boolean invoked, Boolean indexed)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindExpressionStatement(CSharpSyntaxNode node, ExpressionSyntax syntax, Boolean allowsAnyExpression, DiagnosticBag diagnostics)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindStatement(StatementSyntax node, DiagnosticBag diagnostics)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindBlock(BlockSyntax node, DiagnosticBag diagnostics, Binder blockBinder)
   bei Microsoft.CodeAnalysis.CSharp.MethodCompiler.BindMethodBody(MethodSymbol method, TypeCompilationState compilationState, DiagnosticBag diagnostics, Boolean generateDebugInfo, ConsList`1& debugImports)
   bei Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileMethod(MethodSymbol methodSymbol, Int32 methodOrdinal, ProcessedFieldInitializers& processedInitializers, SynthesizedSubmissionFields previousSubmissionFields, TypeCompilationState compilationState)
   bei Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileNamedType(NamedTypeSymbol symbol)
   bei Microsoft.CodeAnalysis.CSharp.MethodCompiler.<>c__DisplayClass4.<CompileNamedTypeAsTask>b__5()
Stapel:
   bei System.Environment.FailFast(System.String, System.Exception)
   bei Microsoft.CodeAnalysis.FatalError.Report(System.Exception, System.Action`1<System.Exception>)
   bei Microsoft.CodeAnalysis.FatalError.Report(System.Exception)
   bei Microsoft.CodeAnalysis.CSharp.MethodCompiler+<>c__DisplayClass4.<CompileNamedTypeAsTask>b__5()
   bei Microsoft.CodeAnalysis.CSharp.BoundFieldAccess.NeedsByValueFieldAccess(Microsoft.CodeAnalysis.CSharp.BoundExpression, Microsoft.CodeAnalysis.CSharp.Symbols.FieldSymbol)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindFieldAccess(Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode, Microsoft.CodeAnalysis.CSharp.BoundExpression, Microsoft.CodeAnalysis.CSharp.Symbols.FieldSymbol, Microsoft.CodeAnalysis.DiagnosticBag, Microsoft.CodeAnalysis.CSharp.LookupResultKind, Boolean)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindNonMethod(Microsoft.CodeAnalysis.CSharp.Syntax.SimpleNameSyntax, Microsoft.CodeAnalysis.CSharp.Symbol, Microsoft.CodeAnalysis.DiagnosticBag, Microsoft.CodeAnalysis.CSharp.LookupResultKind, Boolean)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindIdentifier(Microsoft.CodeAnalysis.CSharp.Syntax.SimpleNameSyntax, Boolean, Microsoft.CodeAnalysis.DiagnosticBag)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindLeftOfPotentialColorColorMemberAccess(Microsoft.CodeAnalysis.CSharp.Syntax.ExpressionSyntax, Microsoft.CodeAnalysis.DiagnosticBag)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindMemberAccess(Microsoft.CodeAnalysis.CSharp.Syntax.MemberAccessExpressionSyntax, Boolean, Boolean, Microsoft.CodeAnalysis.DiagnosticBag)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindInvocationExpression(Microsoft.CodeAnalysis.CSharp.Syntax.InvocationExpressionSyntax, Microsoft.CodeAnalysis.DiagnosticBag)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindExpressionInternal(Microsoft.CodeAnalysis.CSharp.Syntax.ExpressionSyntax, Microsoft.CodeAnalysis.DiagnosticBag, Boolean, Boolean)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindExpression(Microsoft.CodeAnalysis.CSharp.Syntax.ExpressionSyntax, Microsoft.CodeAnalysis.DiagnosticBag, Boolean, Boolean)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindExpressionStatement(Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode, Microsoft.CodeAnalysis.CSharp.Syntax.ExpressionSyntax, Boolean, Microsoft.CodeAnalysis.DiagnosticBag)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindStatement(Microsoft.CodeAnalysis.CSharp.Syntax.StatementSyntax, Microsoft.CodeAnalysis.DiagnosticBag)
   bei Microsoft.CodeAnalysis.CSharp.Binder.BindBlock(Microsoft.CodeAnalysis.CSharp.Syntax.BlockSyntax, Microsoft.CodeAnalysis.DiagnosticBag, Microsoft.CodeAnalysis.CSharp.Binder)
   bei Microsoft.CodeAnalysis.CSharp.MethodCompiler.BindMethodBody(Microsoft.CodeAnalysis.CSharp.Symbols.MethodSymbol, Microsoft.CodeAnalysis.CSharp.TypeCompilationState, Microsoft.CodeAnalysis.DiagnosticBag, Boolean, Roslyn.Utilities.ConsList`1<Microsoft.CodeAnalysis.CSharp.Imports> ByRef)
   bei Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileMethod(Microsoft.CodeAnalysis.CSharp.Symbols.MethodSymbol, Int32, ProcessedFieldInitializers ByRef, Microsoft.CodeAnalysis.CSharp.SynthesizedSubmissionFields, Microsoft.CodeAnalysis.CSharp.TypeCompilationState)
   bei Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileNamedType(Microsoft.CodeAnalysis.CSharp.Symbols.NamedTypeSymbol)
   bei Microsoft.CodeAnalysis.CSharp.MethodCompiler+<>c__DisplayClass4.<CompileNamedTypeAsTask>b__5()
   bei System.Threading.Tasks.Task`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].InnerInvoke()
   bei System.Threading.Tasks.Task.Execute()
   bei System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   bei System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   bei System.Threading.Tasks.Task.ExecuteWithThreadLocal(System.Threading.Tasks.Task ByRef)
   bei System.Threading.Tasks.Task.ExecuteEntry(Boolean)
   bei System.Threading.ThreadPoolWorkQueue.Dispatch()

```

**Comment by pdelvo** [comment]
I tried a newer version of the compiler. The bug us already resolved in e561c8ca

**Comment by gafter** [comment]
Closed as this appears to be fixed as described above by @pdelvo
